### PR TITLE
refactor: Optimizing the PhotoList component to have a single Dimensi…

### PR DESCRIPTION
**Pull request by kosssi**
There is a problem with the Dimension component that adds a div with the following style for each PhotoList (I think it's not necessary):

    overflow: visible; height: 0px; width: 0px;

So I prefer create an PhotoLists component with only one Dimension.

### CODE REVIEW

**Comment by gregorylegarec** [comment]
It looks ok, but shouldn't this component be in a dedicated file instead of exporting two component in `PhotoLists` ?

File: src/components/PhotoList.jsx [hunk 1]
```diff
@@ -57,4 +57,24 @@ export class PhotoList extends Component {
...
 }
 
-export default Dimensions()(PhotoList)
+export class PhotoLists extends Component {
+  render () {
+    const { photoLists, containerWidth, selected, onPhotoToggle } = this.props
+    return (
+      <div>
+        { photoLists.map(photoList => {
+          return (<PhotoList
```
Comment (enguerran) [inline src/components/PhotoList.jsx]: Would be clearer if you use arrow function's shortcuts:

```js
render () {
  const { photoLists, containerWidth, selected, onPhotoToggle } = this.props
  return (
    <div>
      { photoLists.map((photoList) => (
        <PhotoList
          key={photoList.title}
          // etc.
        />
      )}
    </div>
  )
}
```

File: src/containers/PhotoBoard.jsx [hunk 2]
```diff
@@ -78,15 +78,9 @@ export class PhotoBoard extends Component {
           <AddToAlbumModal />
         }
         {showSelection && <SelectionBar />}
-        {!isBusy && photoLists.map(photoList => {
-          return (<PhotoList
```
Comment (enguerran) [inline src/containers/PhotoBoard.jsx]: It seems the previous comment takes his origin here, as a copy/paste refacto. 🍷

File: src/containers/PhotoBoard.jsx [hunk 3]
```diff
@@ -78,15 +78,9 @@ export class PhotoBoard extends Component {
...
-            key={photoList.title}
-            title={photoList.title}
-            photos={photoList.photos}
-            selected={selected}
-            onPhotoToggle={onPhotoToggle}
-          />)
-        })}
+        {!isBusy &&
+          <PhotoLists photoLists={photoLists} selected={selected} onPhotoToggle={onPhotoToggle} />
```
Comment (enguerran) [inline src/containers/PhotoBoard.jsx]: As `PhotoLists` don't do anything, maybe it worth to write down:

```js
{ !isBusy &&
  Dimensions()(photoLists.map(...))
}
```

File: src/components/PhotoList.jsx (hunk 1 above)
Comment (enguerran) [inline src/components/PhotoList.jsx]: If `PhotoList` is not aimed to be used outside (instead of `PhotoLists`), it may not be exported.

Comment (gregorylegarec) [inline src/components/PhotoList.jsx]: @enguerran What do you think about the common role of `PhotoLists` and `Photoboard` ? I feel that `PhotoLists` just adds a `div` level into `PhotoBoard`.

Comment (enguerran) [inline src/components/PhotoList.jsx]: @gregorylegarec In that way we must not create a *just add a `div`* component, but refactor the `render` function of the parent component to integrate this div with something like that, for instance:

```js
render () {
  const lists = photoLists.map((photoList) => (<PhotoList key={photoList.title} />))
  const withDimensions = Dimensions()(<div>lists</div>)
  return (
    <div className='photoBoard'>
      {!isBusy && withDimensions}
    </div>
  )
}

File: src/containers/PhotoBoard.jsx [hunk 4]
```diff
@@ -113,4 +115,4 @@ const mapDispatchToProps = (dispatch, ownProps) => ({
 export default connect(
   mapStateToProps,
   mapDispatchToProps
-)(translate()(PhotoBoard))
+)(translate()(Dimensions()(PhotoBoard)))
```
Comment (enguerran) [inline src/containers/PhotoBoard.jsx]: Regarding previous discussion, it seems legit to `Dimensions()` the `PhotoBoard`, what do you think @gregorylegarec ?

Comment (gregorylegarec) [inline src/containers/PhotoBoard.jsx]: @enguerran That's was my point, yes.

**Comment by gregorylegarec** [comment]
Looks great to me 👍

**Pull request merged**